tests/unit/tracing_test.cpp

```diff
@@ -0,0 +1,74 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+
+#include <gtest/gtest.h>
+#include <gmock/gmock.h>
+#include "mocking.h"
+
+using namespace testing;
+
+extern "C" {
+   #include <tilck/kernel/test/tracing.h>
+}
+
+extern bool *traced_syscalls;
+extern char *traced_syscalls_str;
+
+class tracing_mock : public KernelSingleton {
+public:
+   MOCK_METHOD(int, handle_sys_trace_arg, (const char *arg), (override));
+};
+
+TEST(simple_wildcard_match, basic)
+{
+   const char *str1 = "test string";
```

Comment:
Don't name test strings like that, please. It's hard to read for me. Put all the test strings as literal while calling `simple_wildcard_match`. Also, use more than one string to match against.

Reply:
So should i just insert directly in the function calls?

Reply:
Yes, absolutely. It will be way more clear that way.